src/librustc_data_structures/lib.rs

```diff
@@ -89,10 +89,12 @@ pub mod thin_vec;
 pub mod tiny_list;
 pub mod transitive_relation;
 pub use ena::unify;
+mod atomic_ref;
 pub mod fingerprint;
 pub mod profiling;
 pub mod vec_linked_list;
 pub mod work_queue;
+pub use atomic_ref::AtomicRef;
```

Comment:
This re-export does look a little out of place, maybe just make the module public?

Reply:
I don't like the `atomic_ref::AtomicRef` look; the re-export could be moved in the file but I don't think it matters too much.